src/sagemaker/utils.py

```diff
@@ -342,7 +342,11 @@ def repack_model(inference_script, source_directory, model_uri, sagemaker_sessio
 
         dirname = source_directory if source_directory else os.path.dirname(inference_script)
```

Comment:
Out of scope of this PR, but I want to know what the expected behavior should be for the following parameter:

If one specifies:

inference_script = 'path/to/inference.py'
source_dir = None

should we be copying in the entire leaf directory to the path of the script or just the script?

Reply:
I have a test explaining this behavior here https://github.com/mvsusp/sagemaker-python-sdk/blob/72651ad6051eeba9046d5026d15b466aece7d944/tests/unit/test_utils.py#L333

We copy just the script.